### PR TITLE
Werror

### PR DESCRIPTION
**Pull request by rileyberton**
-Werror was not on for libmtev and there were many warnings (mostly copy/paste unused variable).

Pay particular attention to `mtev_net_heartbeat.c` which defined and filled an IPV6 struct but then never used it?

### CODE REVIEW

**Comment by aidancully** [comment]
Can you run `mtev_config --cflags` with this change? I'm worried `-Werror` will be included in the output, and I think it shouldn't be.

**Comment by rileyberton** [comment]
Yes, mtev-config does the wrong thing now.  looking into it.

**Comment by postwait** [comment]
needs work... doing that now.

File: src/modules/mtev_fq.c [hunk 1]
```diff
@@ -323,7 +323,7 @@ fq_driver_config(mtev_dso_generic_t *img, mtev_hash_table *options) {
   void *data;
 
   while(mtev_hash_next(options, &iter, &key, &klen, &data)) {
-    mtevL(mtev_error, "%s %s!\n", key, data);
+    mtevL(mtev_error, "%s %s!\n", key, (char *)data);
```
Comment (aidancully) [inline src/modules/mtev_fq.c]: would it be better to use `mtev_hash_next_str` here, instead of adding a typecast?

**Comment by postwait** [comment]
LGTM

**Pull request merged**